src/sqlfluff/core/config.py

```diff
@@ -797,6 +797,12 @@ def __getstate__(self):
         state = self.__dict__.copy()
         # Remove the unpicklable entries.
         del state["_plugin_manager"]
+        if "templater_obj" in state["_configs"]["core"]:
```

Comment:
Nitpick: IIRC, you could use `pop()` and pass a default value. Same behavior, but just one line of code.

Reply:
Oohhh great idea. Thanks for the tip. I did think my approach was a little clumsy.